jupyterlab/labapp.py

```diff
@@ -536,8 +537,32 @@ class LabApp(NBClassicConfigShimMixin, LabServerApp):
         help="Whether to expose the global app instance to browser via window.jupyterlab",
     )
 
-    collaborative = Bool(
-        False, config=True, help="Whether to enable collaborative mode (experimental)."
+    collaborative = Bool(False, config=True, help="Whether to enable collaborative mode.")
+
+    collaborative_file_poll_interval = Int(
+        1,
+        config=True,
+        help="""The period in seconds to check for file changes in the back-end (relevant only
+        in collaborative mode). Defaults to 1s, if 0 then file changes will only be checked when
+        saving changes from the front-end.""",
+    )
+
+    collaborative_document_cleanup_delay = Int(
+        60,
+        allow_none=True,
+        config=True,
+        help="""The delay in seconds to keep a document in memory in the back-end after all clients
+        disconnect (relevant only in collaborative mode). Defaults to 60s, if None then the
+        document will be kept in memory forever.""",
+    )
+
+    collaborative_ystore_class = Type(
+        default_value=JupyterSQLiteYStore,
+        klass=BaseYStore,
+        config=True,
+        help="""The YStore class to use for storing Y updates (relevant only in collaborative mode).
+        Defaults to JupyterSQLiteYStore, which stores Y updates in a '.jupyter_ystore.db' SQLite
```

Comment:
Once we hook up the file id service to track file and directory moves/renames, I think this DB should not be stored in the local directory, but rather at the root of the file system in an appropriate Jupyter runtime dir.

Reply:
Yes, I agree.